sympy/physics/optics/tests/test_gaussopt.py

```diff
@@ -90,8 +90,14 @@ def test_gauss_opt():
     z, l, w = symbols('z l r', positive=True)
```

Comment:
instead of `k = p.w`, I think it is better to change this to
`z, l, w_0 = symbols('z l w_0', positive = True)`

As far as I understand the `w` in `BeamParameter(l, z, w)` is actually the `w_0` as found commonly in literature. (it just uses the keyword `w` when we instantiate the object)

Reply:
I have made the suggested change ,thanks @ThePauliPrinciple .

Reply:
The names are now in line with your suggestions, is the pr correct now ?

Reply:
The OP still needs updating w.r.t. this topic.

Reply:
Thanks for reminding . I have changed the OP accordingly .I think this pr is very near to completion so I request you to completely review it !

Reply:
@ThePauliPrinciple Would be glad if you can check it out and point if any more changes are required .